src/Fantomas.Tests/CompilerDirectivesTests.fs

```diff
@@ -300,7 +300,7 @@ let x = 1
 """
 
 [<Test>]
-let ``issue 382`` () =
+let ``issue 382 and issue 889`` () =
```

Comment:
Sorry, I made a mistake here.
The pipe, in this case, is necessary, otherwise, the [AST is invalid](https://fsprojects.github.io/fantomas-tools/#/ast?data=N4KABGBEAmCmBmBLAdrAzpAXFAYgQQCEAZAUQH0BhAeQFkAFASVICVIAacKRNHNRLMPACGAGzSwOESAGMA9nAGQALgE8ADrDAUArgCddsZNJVgAvAB1d55BAgB6O2AAqsALZrZuobpNIAHmDayEqIImBCYKgA7s4AFrJKsQB0AFJosshJAHKwSmBqQtIA1kIA5prcYAYisELi0JbWtmAOYADKsJqxSkpqaJgOpYiJ2gBGSXKudjRCyOUiM36IrrB2TvGJdmraIiJ2AOwADI3IAMSI8GD4xOTU9EwkzCfNarqIAG5CSrAnp4bQF2etgo8k0skuaCUbzmkBAAF8gA).

An additional check is necessary if the [bar has trivia content before](https://fsprojects.github.io/fantomas-tools/#/trivia?data=N4KABGBEDGD2AmBTSAuKAXAngB0WAwgK4BOxiAdtJmALwA6xd5EEA9K2ACqIC22sxAIbFqAMwCWADzCFy6cQBswgsOUQB3LgAtY6LQDoAUgGdY5fQDlE6MNkHQA1oIDmeccbBkFiQccTwGJhYwdjAAZUQ8LXR0bGMUdmdxPUIAI304HlYAWUFyVwVcyXEeRFZOHT1WbEIFBVYAdgAGQPIAYnFRMAAxAEEAIQAZAFEAfXwAeWyABQBJEYAlVuDsYnEAN0F0RFa2inhO5eDjk%2BCAHwIEPFgu43Q1-MgAGnAoJAk1Y1QoPqGxyZm82GC2er0g7m6xnE31EggUfhAAF8gA).
Or in case the [UnionCase has content before](https://fsprojects.github.io/fantomas-tools/#/trivia?data=N4KABGBEDGD2AmBTSAuKAXAngB0WAwgK4BOxiAdtJmALwA6xd5EEA9K2ACqIC22sxAIbFqAMwCWADzCFy6cQBswgsOUQB3LgAtY6LQDoAUgGdY5fQDlE6MNkHQA1oIDmeccbBkFiQccTwGJhYwdjAAZUQ8LXR0bGMUdmdxPUIAI304HlYAWUFyVwVcyXEeRFZOHT1WbEIFBVYAdgAGQPIAYnFRMAAxAEEAIQAZAFEAfXwAeWyABQBJEYAlVuDsYnEAN0F0RFa2inhO5eDjk%2BP8BDxYLuN0NfzIABpwKCQJNWNUKD6hscmZ%2BeGC0ez0g7m6xnEn1EggUfhAAF8gA).

I think you can extract the check out of `sepNlnBasedOnTrivia`

Reply:
Reverted the test.
Refactored the test for trivia.

Reply:
Please remove the `low-hanging-fruit` label 😆